lib/ansible/modules/cloud/azure/azure_rm_deployment.py

```diff
@@ -490,6 +490,12 @@ def deploy_template(self):
                 uri=self.template_link
             )
 
+        # this is to fix issue #35536 -- it's impossible to pass int parameters
+        if self.parameters and self.template:
+            for k in self.parameters:
+                if self.template['parameters'][k]['type'] == "int":
```

Comment:
So to understand this correctly, the user must specify the type as an int in the parameter definition like so?

```
- azure_rm_deployment:
    template:
      parameters:
        intFieldParameter:
          type: int
          value: 123
```